hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -532,8 +532,26 @@ String getDescriptor() {
 			return fieldDescription.getDescriptor();
 		}
 
-		boolean isVisibleTo(TypeDescription typeDescription) {
-			return fieldDescription.isVisibleTo( typeDescription );
+		boolean isVisibleTo(TypeDescription type) {
+			final var declaringType = fieldDescription.getDeclaringType().asErasure();
+			if ( declaringType.isVisibleTo( type ) ) {
+				if ( fieldDescription.isPublic() ) {
+					return true;
+				}
+				else if ( type.equals( declaringType ) ) {
+					return true;
+				}
+				else if ( fieldDescription.isProtected() ) {
+					return declaringType.isAssignableFrom( type );
+				}
+				else if ( fieldDescription.isPrivate() ) {
+					return type.isNestMateOf( declaringType );
+				}
+				// We explicitly consider package-private fields as not visible, as the classes
+				// might have the same package name but be loaded by different class loaders.
+				// (see https://hibernate.atlassian.net/browse/HHH-19784)
```

Comment:
I _think_ we don't have a test for that, correct? That's fine, but I just wanted to check that it's simply because it's too hard to set up.

Reply:
You are correct, it's quite hard to test mappings in different EARs or even EJB jars using separate class loaders - we tried in the past and didn't get anywhere. We might give it another try, but it might be much simpler to do in something like a Wildfly integration test.